src/backend/cuda/platform.cpp

```diff
@@ -478,6 +598,21 @@ int DeviceManager::setActiveDevice(int device, int nId)
     return old;
 }
 
+void DeviceManager::resetcublasHandle(int device)
+{
+    cublasHandles[device].reset(new cublas::cublasHandle());
+}
+
+void DeviceManager::resetcusolverHandle(int device)
+{
+    cusolverHandles[device].reset(new cusolver::cusolverDnHandle());
+}
+
+void DeviceManager::resetcusparseHandle(int device)
+{
+    cusparseHandles[device].reset(new cusparse::cusparseHandle());
+}
+
```

Comment:
I will go through them again. The sections of code that was managing sparse, solve and blas handles was using boost::scoped_ptr earlier, which i replaced with it's std equivalent unique_ptr.